README
-------------------------------------------------
		 SOFTSUSY
-------------------------------------------------
by: B C Allanach, P Athron, A Bednyakov, M Bernhardt, T Cridge, D Grellscheid,
                  M Hanussek, C H Kom, S Martin, D Robertson, R Ruiz de Austri,
		  P Slavich, L Tunstall, A Voigt and A G Williams 

SUMMARY
-------
This program provides a SUSY spectrum in the NMSSM, or the MSSM including
flavour violation and with or without R-parity consistent with input Standard
Model fermion mass/mixings and electroweak/strong coupling data. The R-parity
violating mode can calculate neutrino masses and mixings to 1 loop. SOFTSUSY
can be used in conjunction with other programs for many different particle
physics calculations. SOFTSUSY now has a mode with 3 loop RGEs and some 2-loop
threshold corrections.  

QUICK INSTALLATION AND RUN TEST
-------------------------------

See the "INSTALL" file in this directory for installation instructions. 

There are six C++ test programs, five of which can be run by the commands 
> ./softsusy.x
> ./rpvsoftsusy.x 
> ./rpvneut.x
> ./softsusy-nmssm.x
> ./higher.x
The output from these commands can be checked against outputTest,
rpvOutputTest, neutOutputTest and respectively.

You can run the SUSY Les Houches Accord input provided by running the commands
> ./softpoint.x leshouches < inOutFiles/lesHouchesInput
> ./softpoint.x leshouches < inOutFiles/slha2Input
> ./softpoint.x leshouches < inOutFiles/rpvHouchesInput
You may check the output of these commands against the output files
lesHouchesOutput, slha2Output and rpvHouchesOutput (all in directory
inOutFiles), respectively, that are provided with the SOFTSUSY distribution

Alternatively, run with command line inputs, for example
> ./softpoint.x nmssm sugra --m0=125 --m12=125 --a0=-300 --tanBeta=10
--lambda=0.1 --lambdaAtMsusy

See the manual (in the doc/ subdirectory) threeLoop.pdf for instructions on
how to switch two-loop threshold corrections/three loop RGEs at run-time, once
they've been compiled. 

INTRODUCTION
------------

The most recent version of SOFTSUSY and full documentation can be obtained
from: http://softsusy.hepforge.org/

This program provides a SUSY spectrum in the NMSSM, or the MSSM including
flavour violation and with or without R-parity consistent with input Standard
Model fermion mass/mixings and electroweak/strong coupling data. The R-parity
violating mode can calculate neutrino masses and mixings to 1 loop. 

REFERENCES
----------

If you use SOFTSUSY to write a paper, please cite (see MCnet guidelines)

[1] B.C. Allanach, Comput. Phys. Commun. 143 (2002) 305-331, hep-ph/0104145,

which is the SOFTSUSY manual for the R-parity conserving MSSM. If you
calculate in the NMSSM, please cite [1] and 
[2] B.C. Allanach, P. Athron, L. Tunstall, A. Voigt and A. Williams,
arXiv:1311.7659. 

If you use the R-parity violating aspects, please cite [1] and
[3] B.C. Allanach and M.A. Bernhardt, Comput. Phys. Commun. 181 (2010) 232,
arXiv:0903.1805. 

If you use it to calculate neutrino masses and mixings, please cite [1], [3] and
[4] B.C. Allanach, M. Hanussek and C.H. Kom, Comput. Phys. Commun. 183 (2012)
785, arXiv:1109.3735. 

If you use the three-loop RGEs or two-loop threshold corrections, please cite
[1] and 
[5] B.C. Allanach, A. Bednyakov and R. Ruiz de Autri, arXiv:1407.6130

If you use the two-loop SUSY QCD corrections to squark and gluino pole masses,
please cite [1] and 
[6] B.C. Allanach, Stephen P. Martin, David G. Robertson and Roberto Ruiz de
Austri, arXiv:1601.06657. 


SOFTSUSY-specific input for SUSY Les Houches Accord input files

Block SOFTSUSY           # SOFTSUSY specific inputs
  1   <TOLERANCE>        # desired fractional accuracy in output
  2   <MIXING>           # quark mixing option
  3   <PRINTOUT>         # gives additional verbose output during calculation
  4   <QEWSB>            # change electroweak symmetry breaking scale
  5   <INCLUDE_2_LOOP_SCALAR_CORRECTIONS>    # Full 2-loop running in RGEs
  6   <PRECISION>        # number of significant figures in SLHA output
  7   <numHiggsLoops>    # number of loops in REWSB/mh calculation
  8   <susyRpvBCatMSUSY> # Switch MSUSY-scale RPV boundary conditions ON
  9   <invertedOutput>   # RPV neutrino output uses normal hierarchy (=0.0) or inverted (=1.0)
 10   <forceSlha1>       # if =1, tries to force output into SLHA *1* format
 11   <m32>              # sets gravitino mass to m32
 12   <printSpectra>     # Prints spectrum even when point is theoretically excluded if=1
 13   <mAFlag>           # If=0 (default), sets tachyonic mA=0, otherwise mA=sqrt(|mA|^2)
 15   <NMSSMTools>       # If=1, enables NMSSMTools compatible SLHA2 output
 16   <MICROMEGAS>       # Micromegas options for NMSSMTools use: 1=RD, 2=DD, 3=ID, 4=both
 17   <NMSDECAY>         # If=1, flags for sparticle decays to be calculated via NMSDECAY
 18   <SoftHiggsOut>     # If=1, then the EWSB conditions output soft Higgs masses in NMSSM
 19   <threeLoopRGEs>    # If=1, then 3-loop MSSM RGEs included (default of 0 to disable)
 20   <gyThresholds>     # If>0, switch on gauge/Yukawa two-loop thresholds (see manual [5] for details). 
                           If=31, they all are switched on (default 0 to disable).
 22    <2-loop squark/gluino>   # Include 2-loop terms in gluino/squark masses (default of 0 to disable)
 23    <expandAroundGluinoPole> # sets expandAroundGluinoPole parameter (default 3)

HIGH ORDERS MODE
----------------
If the 2-loop SUSY QCD corrections to squark and gluino masses are required, do
> ./configure --enable-two-loop-sparticle-mass-compilation
> make 
An example point using the higher order terms can be run with, for example,
> ./softpoint.x sugra --tol=1.0e-4 --m0=1000 --m12=1000 --a0=0 --tanBeta=10 --sgnMu=1 --two-loop-sparticle-masses --two-loop-sparticle-mass-method=1

HIGH ACCURACY MODE
------------------
If the high accuracy mode with 3-loop RGEs and some 2-loop threshold corrections is required, do
> ./configure --enable-full-susy-threshold-compilation --enable-three-loop-rge-compilation
> make 
An example point using the high accuracy mode can be run with, for example,
> ./softpoint.x sugra --tol=1.0e-5 --m0=7240 --m12=800 --a0=-6000 --tanBeta=50 --sgnMu=1 --mt=173.2 --alpha_s=0.1187 --mbmb=4.18 --two-loop-susy-thresholds --three-loop-rges
See [5] for more details.

There are detailed comparisons between SOFTSUSY and other publicly available
codes in 
        * Precise Determination of the Neutral Higgs Boson Masses in the MSSM,
	B.C. Allanach, A. Djouadi, J.L. Kneur, W. Porod, P. Slavich, JHEP 0409
	(2004) 044, hep-ph/0406166, 
        * Theoretical uncertainties in sparticle mass predictions from
	computational tools, B.C. Allanach, S. Kraml, W. Porod, JHEP 03 (2003)
	045, hep-ph/0302102  

EXECTUABLE FILES: AFTER INSTALLATION
------------------------------------
    * softpoint.x: command-line interface. GMSB, AMSB, mSUGRA and general
    boundary conditions possible. Main program: src/softpoint.cpp
    * softsusy.x: example C++ test program - calculates spectrum of SPS1a
    mSUGRA point with varying tan beta. Main program: src/main.cpp
    * softsusy-nmssm.x: example NMSSM test program - loops over tan beta. Main
    program: src/main-nmssm.cpp
    * rpvsoftsusy.x: example C++ test program - calculates spectrum of SPS1a
    mSUGRA point with varying lambda'_{331}(M_GUT). Main program: 
    src/rpvmain.cpp
    * rpvneut.x: example neutrino mass calculating R-parity violating 
    test program. 
    * higher.x: example program to illustrate 2-loop QCD corrections to	
    squark and gluino pole masses (note - you must install NLLFAST).

FILES INCLUDED IN THIS DISTRIBUTION
-----------------------------------
Source files are to be found in the src/ subdirectory.

INPUT AND INFO FILES
--------------------
    * [README] contains installation instructions and a list of included files
    * [inOutFiles/outputTest] is the output from the test program
    * [inOutFiles/slha2Input] is an alternative input file in the SUSY Les
    Houches Accord 
    2 format for SPS1a' 
    * [inOutFiles/slha2Output] is the result of running with the above input
    file and 
    includes flavour violation, for inclusion into codes like SusyBsg1.3 which
    include flavour corrections     
    * [inOutFiles/lesHouchesInput] is an alternative input file in the SUSY  
    Les Houches Accord format
    * [inOutFiles/rpvHouchesInput] is an alternative input file in the SUSY 
    Les Houches Accord format for R-parity violation
    * [inOutFiles/rpvHouchesOutput] is the output from the R-parity violating
    test program rpvmain.cpp
    
LINK TO NMSSMTOOLS
------------------

This feature is now deprecated, but I have included instructions for SOFTSUSY
authors. Users may ignore this. 
To link with NMSSMTools, so that SOFTSUSY NMSSM output can be piped through it
(to calculate decays, for example), you'll need to download 
NMSSMTools4.2.1. Then, you can link it using by running 
> ./setup_nmssmtools.sh --nmssmtools-dir=/path/to/NMSSMTools --compile 
After this, there will be an additional executable ./softsusy_nmssmtools.x
that calculates the spectrum with SOFTSUSY, and decays with NMSSMTools. It can
be run by (for example): 
> ./softsusy_nmssmools.x leshouches < inOutFiles/nmssmSLHAnoZ3Input 
If you use it, you should of course provide the proper citations to
NMSSMTools. Using the program in this mode, you should make sure in the SLHA
input file that Block SOFTSUSY variables 15 and 17 are set to 1.
Other files that you need (and are not currently in the SOFTSUSY public
distribution) are: setup_nmssmtools.sh, softsusy_nmssmtools.x.in,
NMSSMTools_4.2.1.tgz, Makefile.nmssmtools


LICENSE
-------

    SOFTSUSY Copyright (C) 2007 B.C. Allanach

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    See http://www.gnu.org/licenses/.  



CHANGES
<li>Removed latex commands from make and improved README.md



main.h

/** \file main.h
   - Project:     SOFTSUSY 
   - Author:      Ben Allanach 
   - Manual:      hep-ph/0104145, Comp. Phys. Comm. 143 (2002) 305 
   - Webpage:     http://hepforge.cedar.ac.uk/softsusy/

   \brief a main C++ program to calculate Higgs masses as a function of tan
   beta 
*/

/** \mainpage Detailed SOFTSUSY Documentation 

    \section install Installation or downloads
    For installation instructions or a download, please go to the 
    <a href="http://projects.hepforge.org/softsusy/">
    SOFTSUSY Homepage</a>

    \section manual Official manual
    If you use SOFTSUSY to write a paper, please cite 
    <a href="http://xxx.arxiv.org/abs/hep-ph/0104145">
    B.C. Allanach, Comput. Phys. Commun. 143 (2002) 305-331, hep-ph/0104145, 
    </a> which is the SOFTSUSY manual.
    If you use the R-parity violating aspects, please cite
    <a href="http://arxiv.org/abs/0903.1805">B.C. Allanach and M.A. Bernhardt, 
    Comput. Phys. Commun. 181 (2010) 232-245,
    arXiv:0903.1805</a>.
    For RPV neutrino oscillation output, please cite 
    <a href="http://xxx.soton.ac.uk/abs/1109.3735">B.C. Allanach,
    M. Hanussek and C.H. Kom, Comput. Phys. Commun. 183 (2012) 785,
    arXiv:1109.3735</a>. 

    \section documentation Documentation
    These web-pages contain the documentation of the latest SOFTSUSY code.
    There are class diagrams and cross-referenced links a la doxygen to help 
    you navigate.

    \section updates Official Updates
    Updates will be posted on the    
    <a href="http://projects.hepforge.org/softsusy/">
    SOFTSUSY Homepage</a>, and the manuals in the doc subdirectory
    will also be updated and released with the distribution.
 */

#include <iostream>
#include <mycomplex.h>
#include <def.h>
#include <linalg.h>
#include <lowe.h>
#include <rge.h>
#include <softsusy.h>
#include <softpars.h>
#include <susy.h>
#include <utils.h>
#include <numerics.h>
#include <rpvsoft.h>
using namespace softsusy;

namespace NR {
  extern int nn;
  extern DoubleVector fvec;
}

  extern void (*nrfuncv)(int n, DoubleVector v, DoubleVector & f);

